torchtune/utils/metric_logging.py

```diff
@@ -317,3 +317,127 @@ def close(self) -> None:
         if self._writer:
             self._writer.close()
             self._writer = None
+
+
+class CometLogger(MetricLoggerInterface):
+    """Logger for use w/ Comet (https://www.comet.com/site/).
+    For more information about arguments expected by Comet, see
+    https://www.comet.com/docs/v2/guides/experiment-management/configure-sdk/#for-the-experiment.
+
+    Args:
+        api_key (Optional[str]): Comet API key. It's recommended to configure the API Key from the environment.
+        workspace (Optional[str]): Comet workspace name. If not provided, uses the default workspace.
+        project (Optional[str]): Comet project name. Defaults to Uncategorized.
+        experiment_name (Optional[str]): The name for comet experiment to be used for logging.
+        experiment_key (Optional[str]): The key for comet experiment to be used for logging. Must be an alphanumeric
+            string whose length is between 32 and 50 characters.
+        mode (Optional[str]): Control how the Comet experiment is started. "get": Continue logging to an existing
+            experiment identified by the `experiment_key` value. "create": Always creates of a new experiment, useful
+            for HPO sweeps. "get_or_create" (default): Starts a fresh experiment if required, or persists logging to
+            an existing one.
+        online (Optional[bool]): If True, the data will be logged to Comet server, otherwise it will be stored locally
+            in offline experiment. Default is `True`.
+        experiment_name (Optional[str]): Name of the experiment. If not provided, Comet will auto-generate a name.
+        tags (Optional[List[str]]): Tags to associate with the experiment.
+        log_code (bool): Whether to log the source code. Defaults to True.
+        **kwargs: additional arguments to pass to Comet.start. See
+            https://www.comet.com/docs/v2/api-and-sdk/python-sdk/reference/Experiment-Creation/#comet_ml.ExperimentConfig
+
+    Example:
+        >>> from torchtune.utils.metric_logging import CometLogger
+        >>> logger = CometLogger(project_name="my_project", workspace="my_workspace")
+        >>> logger.log("my_metric", 1.0, 1)
+        >>> logger.log_dict({"my_metric": 1.0}, 1)
+        >>> logger.close()
+
+    Raises:
+        ImportError: If `comet_ml` package is not installed.
+
+    Note:
+        This logger requires the comet_ml package to be installed.
+        You can install it with ``pip install comet_ml``.
+        You need to set up your Comet.ml API key before using this logger.
+        You can do this by setting the COMET_API_KEY environment variable
+        or by calling ``comet_ml.login()`` with your API key.
```

Comment:
personally I find comet login the easiest to do and is more comfortable for folks who don't know much code. What are your thoughts on just always recommending to setup via comet login so folks do not get confused with the different options?

Reply:
`comet login` works great in interactive situations but won't help in automated setups. I'm happy to highlight the `comet login` option and links to other options in the Comet docs if that works for you.

Reply:
Same comment as in https://github.com/pytorch/torchtune/pull/1221#discussion_r1710221128, if we expect users to be setting up a config and doing `tune run ...` how are they supposed to use `comet.login()`? We would have to call that inside the logger, and incorporate some API key via a secrets manager solution which would be different for every user.

As far as I can tell torchtune seems to heavily favor the CLI `tune run` style usage, rather than importing it and calling tune run programmatically. If this is indeed the case, I think we should actually remove the mention to `comet.login()` and have people export the api key, what do you guys think?

Reply:
Sorry I was under the impression you can run `comet login` from CLI, is this not the case?

Either way I agree with @dzheng256, since CLI is the primary entry point for most users via `tune run` I would lean towards suggesting whatever fits that workflow best whether that's setting environment variables or a CLI command

Reply:
@RdoubleA @dzheng256 Sorry about the confusion, there is bot a `comet login` command-line script and `comet_ml.login`  that one can call in their Python script. I will clean up and double-check everything to point people to use the CLI `comet login`.